File: apps/seahax/api/sentry/sentry.go
package sentry

import (
	"fmt"
	"log/slog"
	"os"
	"seahax/api/config"
	"sync"

	"github.com/getsentry/sentry-go"
)

var Get = sync.OnceValue(func() *sentry.Client {
	config := config.Get()
	err := sentry.Init(sentry.ClientOptions{
		Dsn:            config.SentryDSN,
		Environment:    config.Environment,
		SendDefaultPII: true,
	})

	if err != nil {
		slog.Error(fmt.Sprintf("sentry init failed: %s", err))
		os.Exit(1)
	}

	slog.Debug("sentry initialized successfully")

	return sentry.CurrentHub().Client()
})

File: apps/seahax/frontend/src/init/sentry.ts
import * as sentry from '@sentry/react';

sentry.init({
  dsn: 'https://c7a75ff7a98d2ca889b5fb5c549046ed@o4509086879973376.ingest.us.sentry.io/4509087149326336',
  environment: window.location.hostname === 'seahax.com' ? 'production' : 'development',
  sendDefaultPii: true,
});
